### ISSUE TITLE
Some functionalities broken since 0.2.0

### ISSUE THREAD

**Issue by camfrout** [innoverio/vscode-dbt-power-user#17]
Hi,

The extension was working fine in 0.1.4 but since the update, to 0.2.0 (and also in 0.2.1) the following is not working:
- Showing parent/children models (is always empty)
- Go to definition
- Autocompletion

This is still working:
- The "play" button to run the current model
- The snippets (typing ref suggests {{ref('')}}

This time, I have an error appearing in the Dev Tools:
```
We should never come here, means that our exceptions are not handled! (at DBTManifest.<anonymous> (/Users/xxx/.vscode-insiders/extensions/innoverio.vscode-dbt-power-user-0.2.1/out/dbtManifest.js:64:25))
```

Switching back to 0.1.4 makes everything work fine again.

**Comment by mdesmet** [comment]
This means  that the manifest is not able to be parsed, there is a unhandled exception in the refresh of the manifest, however we don't get the actual error as we forgot to log it.

I'm not able to reproduce your issue with latest version. Did a diff between the versions, but no immediate clue on what's going.

I will make an update to add this logging, would you try again?

In the future we will need to add telemetry so exceptions are reported back to us.

**Comment by camfrout** [comment]
Of course I am happy to help debugging the issue, I really like what your extension is doing!

Here is the error with 0.2.2:
```
We should never come here, means that our exceptions are not handled! TypeError: Cannot convert undefined or null to object
	at Function.values (<anonymous>)
	at DBTManifest.createSourceMetaMap (/Users/xxxx/.vscode-insiders/extensions/innoverio.vscode-dbt-power-user-0.2.2/out/dbtManifest.js:131:16)
	at DBTManifest.<anonymous> (/Users/xxxx/.vscode-insiders/extensions/innoverio.vscode-dbt-power-user-0.2.2/out/dbtManifest.js:89:40)
	at Generator.next (<anonymous>)
	at /Users/xxxx/.vscode-insiders/extensions/innoverio.vscode-dbt-power-user-0.2.2/out/dbtManifest.js:8:71
	at new Promise (<anonymous>)
	at __awaiter (/Users/xxxx/.vscode-insiders/extensions/innoverio.vscode-dbt-power-user-0.2.2/out/dbtManifest.js:4:12)
	at DBTManifest.refresh (/Users/xxxx/.vscode-insiders/extensions/innoverio.vscode-dbt-power-user-0.2.2/out/dbtManifest.js:74:16)
	at DBTManifest.<anonymous> (/Users/xxxx/.vscode-insiders/extensions/innoverio.vscode-dbt-power-user-0.2.2/out/dbtManifest.js:61:28)
	at Generator.next (<anonymous>)
	at /Users/xxxx/.vscode-insiders/extensions/innoverio.vscode-dbt-power-user-0.2.2/out/dbtManifest.js:8:71
	at new Promise (<anonymous>)
	at __awaiter (/Users/xxxx/.vscode-insiders/extensions/innoverio.vscode-dbt-power-user-0.2.2/out/dbtManifest.js:4:12)
	at DBTManifest.tryRefresh (/Users/xxxx/.vscode-insiders/extensions/innoverio.vscode-dbt-power-user-0.2.2/out/dbtManifest.js:59:16)
	at activate (/Users/xxxx/.vscode-insiders/extensions/innoverio.vscode-dbt-power-user-0.2.2/out/extension.js:16:39)
	at Function._callActivateOptional (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:916:841)
	at Function._callActivate (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:916:492)
	at /Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:914:724
	at processTicksAndRejections (internal/process/task_queues.js:94:5)
	at async Promise.all (index 0)
	at async Promise.all (index 84)
	at async Promise.all (index 1) (at DBTManifest.<anonymous> (/Users/xxxx/.vscode-insiders/extensions/innoverio.vscode-dbt-power-user-0.2.2/out/dbtManifest.js:64:25))
```

**Comment by mdesmet** [comment]
Huh, that's interesting, that means there is no "sources" key in the manifest file. Could you validate this? Just search for the following string  **"sources": {** inside the manifest.json file in the target folder.

What version of dbt are you using?

**Comment by camfrout** [comment]
That's it. I am currently using `dbt==0.15.2` due to compatibility issues for later versions with `dbt-sqlserver`

Running `cat target/manifest.json | jq '.sources'` after generating `manifest.json` via dbt 0.15.2 returns NULL.
But if I run the same command with dbt 0.18rc1 it returns some values and the extension works again (but I can't really use 0.18 yet on my project until it is officially released and `dbt-sqlserver` is updated to work with 0.18).

Not sure if it is worth updating the extension or just mentioning in the README that it works only from a specific version of dbt.

**Comment by mdesmet** [comment]
So are you using sources within your project? if so how are they represented n the manifest file? If it's not too hard we can add support.

In all caes the failure of manifest parsing in one part (sources) should not impact the others (models, macros, ...). I am going to push out some fix to provide more isolation.

**Comment by camfrout** [comment]
Yes, we are using sources.

In 0.15.2 the sources for each model are stored under `nodes > [model_name] > sources`. 
And it seems that the list of all sources for a given `yml` file is stored under `files > [file_name] > sources` but I couldn't find a single parameter holding the list of all sources across all `yml` files.

**Comment by camfrout** [comment]
Fixed in 0.2.3. 
Thanks for the super quick turnaround!
